# Metagenomics_Tutorial
Metagenomics is the study of genetic material directly recovered from environmental samples and therefore cultivation is not required.

A metagenome is a collection of genomes from the assemblage of microorganisms present in a defined environment (Handelsman et al. Chemistry Biology, 1998)

To analyze metagenomes several steps are required
1. Quality check and trimming
2. Assembly
3. Binning
6. Annotation

To further improve the quality and completeness of your metagenomic reads you can perform Bin targeted reassemblies. This means you use your retrieved bins and map these back to the assembly. This might fill gaps and lengthen sequences, 



## Quality check and trimming
Removal of redundant, low quality sequences
### FastQC
### Illumina-utils

## Optional: Coverage, Diversity and similarity of metagenomic samples

- [Nonpareil](http://enve-omics.ce.gatech.edu/nonpareil/): a redundancy-based approach to assess the level of coverage in metagenomic datasets
- [Mash](): R (vegan), Pairwise distances, ordination plots 



## Optional: Abundance estimation
### Kraken2/Bracken

## Optional: Reference guided assembly

## Assembly
### Megahit

## Readmapping
### Bowtie2

## Binning
### Anvi'o

## Bin targeted reassemblies
### Spades
